src/compiler/checker.ts

```diff
@@ -12495,6 +12509,36 @@ namespace ts {
         }
 
         /**
+         * Checks that the return type provided is an instantiation of the global Promise<T> type

```

Comment:
The `checker.ts` code for generators uses `checkTypeAssignableTo` to check the return type annotation ([L11625](https://github.com/Microsoft/TypeScript/blob/master/src/compiler/checker.ts#L11625)). Why is `checkTypeAssignableTo` not used for async functions?